AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -621,9 +620,6 @@ final public class AWSMobileClient: _AWSMobileClient {
         self.performHostedUISuccessfulSignInTasks(disableFederation: hostedUIOptions.disableFederation, session: session, federationToken: federationToken!, federationProviderIdentifier: federationProviderIdentifier, signInInfo: &signInInfo)
         self.mobileClientStatusChanged(userState: .signedIn, additionalInfo: signInInfo)
         completionHandler(.signedIn, nil)
-        if self.pendingGetTokensCompletion != nil {
-            self.tokenFetchLock.leave()
-        }
```

Comment:
is there a change in behavior here? we used to enter the lock on getTokens, then check that the refresh token expired, and then set pendingGetTokensCompletion. developers listen to the state for `signedOutUserPoolsTokenInvalid` and will prompt the user to sign back in. When the user signs back in, this method is called. If there is a `pendingGetTokensCompletion` what should we do that is the equivalent to before?  should it be
```
if self.pendingGetTokensCompletion != nil {
            self.pendingGetTokensCompletion("??", nil) // so that the `done()` is called in AWSMobileClientExtensions.swift line 856

```
Amplify's auth plugin will set up the listeners and call releaseSignIn immediately, which will call pendingGetTokensCompletion with an error and set it to `nil`.

Reply:
oh, it's calling it in the next line